### ISSUE TITLE
Split features tool crashes QGIS

### ISSUE THREAD

**Issue by frspp** [qgis/QGIS#52020]
## User Feedback

split features tool crashes.

## Report Details

**Python Stack Trace**
```
Windows fatal exception: access violation

Current thread 0x00004cf0 (most recent call first):
<no Python frame>
```

**Stack Trace**

```

QgsGeos::linePointDifference :
QgsGeos::splitGeometry :
QgsGeometry::splitGeometry :
QgsGeometry::splitGeometry :
QgsVectorLayerEditUtils::splitFeatures :
QgsVectorLayer::splitFeatures :
QgsMapToolSplitFeatures::cadCanvasReleaseEvent :
QgsMapToolAdvancedDigitizing::canvasReleaseEvent :
QgsMapCanvas::mouseReleaseEvent :
QWidget::event :
QFrame::event :
QGraphicsView::viewportEvent :
QCoreApplicationPrivate::sendThroughObjectEventFilters :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify :
QCoreApplication::notifyInternal2 :
QApplicationPrivate::sendMouseEvent :
QSizePolicy::QSizePolicy :
QSizePolicy::QSizePolicy :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify :
QCoreApplication::notifyInternal2 :
QGuiApplicationPrivate::processMouseEvent :
QWindowSystemInterface::sendWindowSystemEvents :
QEventDispatcherWin32::processEvents :
qt_plugin_query_metadata :
QEventLoop::exec :
QCoreApplication::exec :
main :
BaseThreadInitThunk :
RtlUserThreadStart :
```

### Steps to reproduce the issue

Add layer from geopackage. 
Select line feature. 
Choose "split features". Click and split.
Works usually few times, then crashes. Now at least 10 crashes.

### Versions

**QGIS Info**
QGIS Version: 3.28.3-Firenze
QGIS code revision: c12bcb2f76c
Compiled against Qt: 5.15.3
Running against Qt: 5.15.3
Compiled against GDAL: 3.6.2
Running against GDAL: 3.6.2

**System Info**
CPU Type: x86_64
Kernel Type: winnt
Kernel Version: 10.0.22621

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [x] I tried with a new QGIS profile

**Comment by frspp** [comment]
New profile worked better but finally got crash again.

**Comment by roya0045** [comment]
Duplicate of issue that should be fixed now, you can test the main branch or wait for the next point release to see if the issue persist.

**Comment by uclaros** [comment]
Judging by the stack trace, this is most probably fixed by #51797
I'm closing this to keep things tidy, but feel free to open it again if you can eventually reproduce it when 3.30 / 3.28.4 lands.
